### PR TITLE
Feature/12635 fss tooling handle the search api endpoint responses

### PR DESCRIPTION
**Pull request by pratibh14837**
PR contains Implementation and Unit test cases

### CODE REVIEW

File: FileShareClient/FileShareApiClient.cs [hunk 1]
```diff
@@ -99,6 +100,40 @@ public async Task<BatchSearchResponse> Search(string searchQuery, int? pageSize
             }
         }
 
+        public async Task<IResult<BatchSearchResponse>> Search(string searchQuery, CancellationToken cancellationToken, int? pageSize = null, int? start = null)
```
Comment (crdandoukho) [inline FileShareClient/FileShareApiClient.cs:103]: Please change the optional parameters to mandatory
```suggestion
        public async Task<IResult<BatchSearchResponse>> Search(string searchQuery, CancellationToken cancellationToken, int? pageSize, int? start)
```

File: FileShareClient/FileShareApiClient.cs [hunk 2]
```diff
@@ -18,6 +18,7 @@ public interface IFileShareApiClient
     {
         Task<BatchStatusResponse> GetBatchStatusAsync(string batchId);
         Task<BatchSearchResponse> Search(string searchQuery, int? pageSize = null, int? start = null);
+        Task<IResult<BatchSearchResponse>> Search(string searchQuery, CancellationToken cancellationToken, int? pageSize = null, int? start = null);
```
Comment (crdandoukho) [inline FileShareClient/FileShareApiClient.cs:21]: Please make the optional parameters mandatory
```suggestion
        Task<IResult<BatchSearchResponse>> Search(string searchQuery, CancellationToken cancellationToken, int? pageSize, int? start);
```

File: FileShareClient/FileShareApiClient.cs (hunk 1 above)
Comment (crdandoukho) [inline FileShareClient/FileShareApiClient.cs:103]: There is much duplicated code in the two Search methods. Could you please move the duplicated code to a new private method and call it from the Search methods?
I have created gist to explain what I mean: https://gist.github.com/crdandoukho/dbc90d4389f3afd27fc36c58bf6e3b55

**Review by crdandoukho — CHANGES_REQUESTED** [review]
Some changes needed

Because you've added new functionality in a backwards compatible way you need to increment the minor version number of the library. Please do this in azure-pipelines.yml
```
  - name: UKHOAssemblyVersionPrefix
    value: "1.0."
```
change to:
```
  - name: UKHOAssemblyVersionPrefix
    value: "1.1."
```

File: FileShareClient/FileShareApiClient.cs (hunk 1 above)
Comment (pratibh14837) [reply FileShareClient/FileShareApiClient.cs]: resolved

File: FileShareClient/FileShareApiClient.cs (hunk 2 above)
Comment (pratibh14837) [reply FileShareClient/FileShareApiClient.cs]: resolved

File: FileShareClient/FileShareApiClient.cs (hunk 1 above)
Comment (pratibh14837) [reply FileShareClient/FileShareApiClient.cs]: resolve

**Review by crdandoukho — APPROVED** [review]
Looks good. 

Thanks for the changes.

**Pull request merged**